runtime-config.md

```diff
@@ -12,27 +12,28 @@ Presently there are `Prestart`, `Poststart` and `Poststop`.
 Hooks allow one to run code before/after various lifecycle events of the container.
 Hooks MUST be called in the listed order.
 The state of the container is passed to the hooks over stdin, so the hooks could get the information they need to do their work.
+All hooks execute in the host environment (e.g. the same namespace, cgroups, etc. that apply to the host process).
 
 Hook paths are absolute and are executed from the host's filesystem.
 
 ### Pre-start
 
-The pre-start hooks are called after the container process is spawned, but before the user supplied command is executed.
+The pre-start hooks are called [after the container process is spawned, but before the user supplied command is executed](runtime.md#typical-lifecycle).
 They are called after the container namespaces are created on Linux, so they provide an opportunity to customize the container.
 In Linux, for e.g., the network namespace could be configured in this hook.
 
-If a hook returns a non-zero exit code, then an error including the exit code and the stderr is returned to the caller and the container is torn down.
+If a hook returns a non-zero exit code, [then an error including the exit code and the stderr is returned to the caller and the container is torn down](runtime.md#typical-lifecycle).
 
 ### Post-start
 
-The post-start hooks are called after the user process is started.
+The post-start hooks are called [after the user process is started](runtime.md#typical-lifecycle).
```

Comment:
On Mon, Oct 12, 2015 at 04:14:38PM -0700, W. Trevor King wrote:
> Mon, Oct 12, 2015 at 04:03:30PM -0700, Vish Kannan:
> > Not really. If you bind-mount the ns files, you don't need a process
> > to hold the namespaces.
> 
> Oh, that will make my life a lot easier :).  Is that…

And it looks like it is :).  Language supporting this in namespaces(7):

  Bind mounting (see mount(2)) one of the files in this directory to
  somewhere else in the filesystem keeps the corresponding namespace
  of the process specified by pid alive even if all processes
  currently in the namespace terminate.

For example:

  tty1$ unshare --user --uts --map-root-user
  tty1$ hostname alice
  tty1$ echo $$
  25477

  tty2# touch /tmp/uts-alice
  tty2# mount --bind /proc/25477/ns/uts /tmp/uts-alice

  tty1$ exit

  tty2# ps aux | grep 25477 | grep -v grep
  … no hits …
  tty2# nsenter --uts=/tmp/uts-alice hostname
  alice